src/demo/app/shared/example-viewer/example-viewer.component.ts

```diff
@@ -32,24 +32,21 @@ export class ExampleHostDirective {
 	],
 })
 export class ExampleViewerComponent implements OnInit {
-	@Input() example: string;
-
+	example = input<string>();
 	@ViewChild(ExampleHostDirective, { static: true }) exampleHost: ExampleHostDirective;
-
 	title: string;
-
-	constructor(private componentFactoryResolver: ComponentFactoryResolver) {}
+	private componentFactoryResolver = inject(ComponentFactoryResolver);
 
 	get sourcePath() {
```

Comment:
@pavankjadda I think it is better to have sourcePath as readonly signal through computed. 

this way sourcePath get updated everytime the input "example" change  


``` javascript
sourcePath = computed(()=>`https://github.com/ng-select/ng-select/tree/master/src/demo/app/examples/${this.example()}`);
````